Description: Use artifacts versions available in Debian and disable unavailable ones
Author: Miguel Landaeta <miguel@miguel.cc>
Forwarded: no
Last-Update: 2013-09-06

--- a/build.gradle
+++ b/build.gradle
@@ -51,10 +51,13 @@
 def theVendor = 'gpars.org'
 def theTitle = 'GPars: Groovy Parallel Systems'
 
+// We are not building the pdf guides at this moment due to missing dependencies
+/*
 apply {
     from 'gradle/docs.gradle'
     from 'gradle/docsDependencies.gradle'
 }
+*/
 
 task copyDSLDefinitions(type: Copy) {
     into "$buildDir/classes/main"
@@ -129,10 +135,12 @@
     docs group: 'org.codehaus.groovy', name: 'groovy-all', version: project.hasProperty('gpars_groovyVersion') ? gpars_groovyVersion : '2.0.8'
     docs 'org.yaml:snakeyaml:1.12'
     docs 'commons-lang:commons-lang:2.6'
+    /* Disabling the build of pdf guides due to missing dependencies
     docs project.ext.grailsDocs
     docs project.ext.radeox
     docs project.ext.lowagieItext
     docs project.ext.xhtmlRenderer
+    */
     docs 'commons-logging:commons-logging:1.1.1'
     deployerJars "org.apache.maven.wagon:wagon-http-lightweight:2.4"
     cover 'net.sourceforge.cobertura:cobertura:1.9.4.1'
@@ -288,7 +296,9 @@
     overview = new File('overview.html')
 }
 
-task documentation(dependsOn: ['javadoc', 'groovydoc', 'buildGuide', 'pdfGuide'], description: 'Create the API documentation.')
+// Only build javadocs as documentation
+//task documentation(dependsOn: ['javadoc', 'groovydoc', 'buildGuide', 'pdfGuide'], description: 'Create the API documentation.')
+task documentation(dependsOn: ['javadoc', 'groovydoc'], description: 'Create the API documentation.')
 
 task zipDoc(type: Jar, dependsOn: 'documentation') {
     classifier = 'javadoc'
--- a/buildSrc/build.gradle
+++ b/buildSrc/build.gradle
@@ -22,7 +22,8 @@
 
 apply plugin: 'groovy'
 
-apply from: '../gradle/docsDependencies.gradle'
+// Don't build pdf guides due to missing dependencies
+//apply from: '../gradle/docsDependencies.gradle'
 
 repositories {
   mavenCentral()
@@ -33,8 +34,10 @@
 dependencies {
 	compile localGroovy()
   compile gradleApi()
+  /* Disabling the building of pdf guides due to missing dependencies
   compile(project.ext.grailsDocs) { transitive = false }
   compile(project.ext.xhtmlRenderer) { transitive = false }
   compile(project.ext.lowagieItext) { transitive = false }
   compile(project.ext.radeox) { transitive = false }
+  */
 }
--- a/buildSrc/src/main/groovy/PdfGuide.groovy
+++ b/buildSrc/src/main/groovy/PdfGuide.groovy
@@ -24,7 +24,8 @@
 import org.gradle.api.tasks.Input
 import org.gradle.api.tasks.OutputDirectory
 import org.gradle.api.tasks.TaskAction
-import grails.doc.PdfBuilder
+// Grails isn't available in Debian yet
+//import grails.doc.PdfBuilder
 
 class PdfGuide extends DefaultTask {
 
@@ -39,11 +40,13 @@
     @TaskAction
     def publish() {
         try {
+            /* Grails isn't available in Debian yet
             PdfBuilder.build(
                     basedir: outputDirectory.absolutePath,
                     home: project.file('grails-doc').absolutePath,
                     tool: 'pdf/gpars'
             )
+            */
         } catch (ignore) {
             // it's very likely that the stream is closed before
             // the renderer 'finishes' but it actually does
